### PR TITLE
Use aws-sdk-go for s3 access

### PR DESCRIPTION
**Pull request by forklady42**
Tests now succeed locally; still need to add expected environment variables to testing environment. Also, should rebase again and squash some of these commits. Plus, there's a bit of refactoring to do, which I may get to this afternoon. Those qualifications aside, this is the bulk of the changes.

r? @colinmarc

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -109,26 +110,22 @@ func localSetup(localPath string, config sequinsConfig) *sequins {
...
-	if !exists {
-		log.Fatalf("Invalid AWS region: %s", regionName)
-	}
+	// Requiring region name in config for now
+	// if regionName == "" {
+	// 	regionName = aws.InstanceRegion()
+	// 	if regionName == "" {
+	// 		log.Fatal("Unspecified S3 region, and no instance region found.")
+	// 	}
+	// }
```
Comment (colinmarc) [inline main.go]: Can we default to the instance region, still? I'm not sure without looking but I think this'll break our deployment on ec2.

File: s3_sequins_test.go [hunk 2]
```diff
@@ -73,7 +70,46 @@ func TestS3Backend(t *testing.T) {
...
+
+	_, err = svc.PutObject(params)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func putS3Blob(svc *s3.S3, bucket, dst string, src []byte) error {
```
Comment (colinmarc) [inline s3_sequins_test.go]: If you make this take an `io.ReadSeeker` or whatever `PutObjectInput` wants, then you can reuse it in the above method.

**Comment by colinmarc** [comment]
(I just added a commit vendoring aws-sdk-go)

File: main.go (hunk 1 above)
Comment (forklady42) [inline main.go]: Ah, good point! I've now updated this code to use aws-sdk-go.

**Comment by forklady42** [comment]
Ok, I've done some tidying and have added the test environment variables to travis. Ready for review.

**Comment by colinmarc** [comment]
One request - can you rebase everything except the vendor commit into one commit? Github won't let me view the total diff, but seeing the other changes commit-by-commit is hard.

**Comment by colinmarc** [comment]
LGTM!!!!!

**Pull request merged**